Donation-platform/src/main/java/com/google/servlets/BrowseCharitiesServlet.java

```diff
@@ -49,9 +49,9 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         if (!isTagName) {
             try
             {
-                Collection<Charity> charityCollection = dbCalls.getAllCharities();
```

Comment:
I realized that this implementation could be adding to the latency issues. I realized we did not need all the charities since on the browse page we display the tags and the tags under the charities. Therefore there's no use of getting everything. That's what I decide to optimize by getting only the tags and charities under them.

Reply:
sg